src/main/java/org/testng/TestNG.java

```diff
@@ -875,9 +875,7 @@ private void initializeCommandLineSuitesGroups() {
     }
   }
   private void addReporter(Class<? extends IReporter> r) {
-    if (! m_reporters.contains(r)) {
-      m_reporters.add(ClassHelper.newInstance(r));
-    }
+    m_reporters.add(ClassHelper.newInstance(r));
```

Comment:
It doesn't sound good: you will create a new instance of r, even if r already exists.